subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -525,129 +435,103 @@ impl Session<Error> for ElectrumSession {
+                }
+                NetworkId::Elements(network) => {
+                    let verifier = Verifier::new(network);
+                    ChainOrVerifier::Verifier(verifier)
                 }
+            };
 
-                loop {
-                    if r.try_recv().is_ok() {
+            let mut headers = Headers {
+                store: store.clone(),
+                checker,
+            };
+
+            let headers_url = self.url.clone();
+            let (close_headers, r) = channel();
+            self.closer.senders.push(close_headers);
+            let mut chunk_size = DIFFCHANGE_INTERVAL as usize;
+            let headers_handle = thread::spawn(move || {
+                info!("starting headers thread");
+
+                'outer: loop {
+                    if wait_or_close(&r, sync_interval) {
                         info!("closing headers thread");
-                        break 'outer;
+                        break;
                     }
-                    match headers_kind.ask(chunk_size) {
-                        Ok(headers_found) => {
-                            if headers_found == 0 {
-                                chunk_size = 1;
-                                break;
-                            } else {
-                                info!("headers found: {}", headers_found);
+
+                    if let Ok(client) = headers_url.build_client() {
+                        loop {
+                            if r.try_recv().is_ok() {
+                                info!("closing headers thread");
+                                break 'outer;
                             }
-                        }
-                        Err(Error::InvalidHeaders) => {
-                            // this should handle reorgs and also broke IO writes update
-                            if headers_kind.remove(144).is_err() {
-                                break;
+                            match headers.ask(chunk_size, &client) {
+                                Ok(headers_found) => {
+                                    if headers_found == 0 {
+                                        chunk_size = 1
+                                    } else {
+                                        info!("headers found: {}", headers_found);
+                                    }
+                                }
+                                Err(Error::InvalidHeaders) => {
+                                    // this should handle reorgs and also broke IO writes update
+                                    if headers.remove(144).is_err() {
+                                        break;
+                                    }
+                                }
+                                Err(e) => {
+                                    // usual error is because I reached the tip, trying asking half
+                                    //TODO this is due to an esplora electrs bug, according to spec it should
```

Comment:
This was fixed, the `chunk_size` can just always be set to `DIFFCHANGE_INTERVAL`